engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -478,26 +478,36 @@ class WorkflowActor(workflowToStart: WorkflowToStart,
     goto(finalState) using data.copy(currentLifecycleStateActor = None)
   }
 
-  private[workflow] def makeFinalizationActor(workflowDescriptor: EngineWorkflowDescriptor, jobExecutionMap: JobExecutionMap, workflowOutputs: CallOutputs) = {
-    val copyWorkflowOutputsActorProps = stateName match {
+  private[workflow] def makeFinalizationActor(workflowDescriptor: EngineWorkflowDescriptor, jobExecutionMap: JobExecutionMap, workflowOutputs: CallOutputs): ActorRef = {
+    val actorsPropsOptionList: Option[List[Props]] = stateName match {
       case InitializingWorkflowState => None
-      case _ => Option(CopyWorkflowOutputsActor.props(workflowIdForLogging, ioActor, workflowDescriptor, workflowOutputs, stateData.initializationData))
+      case _ => Option(List(CopyWorkflowOutputsActor.props(workflowIdForLogging, ioActor, workflowDescriptor, workflowOutputs, stateData.initializationData),
+        CopyWorkflowMetadataActor.props(workflowIdForLogging, ioActor, serviceRegistryActor, workflowDescriptor, workflowOutputs, stateData.initializationData)))
     }
-    
+
+    val actorsList: List[Props] = actorsPropsOptionList.getOrElse(List(Props.empty))
```

Comment:
I think it would be better to add some context here. Maybe keep these actors separately or create a case class for wrapping them?

Reply:
yeah, that worth being done